### PR TITLE
Fix lockup in Linux when AEM status return is not successful

### PR DESCRIPTION
**Pull request by ahogen**
Fixes audioscience/avdecc-lib#350.

(Apologies for the unnecessary removal of blank lines in `cmd_line.cpp`. Something I didn't realize XED was doing until I saw the diff here.)

### CODE REVIEW

**Comment by andrew-elder** [comment]
Looks good to me. @vronchetti - what do you think?

**Comment by ahogen** [comment]
I saw this if statement wasn't in some of the original code so I went digging for a while and found that it was introduced into the master branch fairly recently. 

- Introduced [here](https://github.com/audioscience/avdecc-lib/commit/b7ea336c37441726d55ddcfa811af90ee9779e61#diff-611aa11172375fb789418ec7931a3399R302)
- Modified (slightly) [here](https://github.com/audioscience/avdecc-lib/commit/524e6c7932e49015162ecae528a348fe22d38ee0#diff-611aa11172375fb789418ec7931a3399L302)

Will this change affect the XMOS firmware upgrade functionality that `fchin` was trying to fix? I don't have an XMOS with firmware upgrade enabled at the moment so I can't verify on my end.

Might be worth testing before this change propagates too far.

**Comment by andrew-elder** [comment]
I think the "Modified" is ok - it is just #define name changes.

The timeout handling for XMOS looks a little tricker to evaluate. Any XMOS guys following this thread?

**Comment by ahogen** [comment]
Oops. Didn't realize new commits would hit this PR. I'll refrain from pushing to `staging` and keep to my own dev branch for now until this XMOS firmware update issue is resolved and/or you pull this PR. Apologies.

File: controller/app/cmdline/src/cmd_line.cpp [hunk 1]
```diff
@@ -1382,30 +1382,34 @@ int cmd_line::do_select(uint32_t new_end_station, uint16_t new_entity, uint16_t
...
-            atomic_cout << "New setting" << std::endl;
-            atomic_cout << "\tEnd Station Index: " << std::dec << current_end_station << " (" << end_station_name << ")" << std::endl;
-            atomic_cout << "\tEntity Index: " << std::dec << end_station->get_current_entity_index() << std::endl;
-            atomic_cout << "\tConfiguration Index: " << std::dec << end_station->get_current_config_index() << std::endl;
+            atomic_cout << "New selection" << std::endl;
+            atomic_cout << "\tEnd Station Index:   " << std::dec << current_end_station << std::endl;
+            atomic_cout << "\tEnd Station:         " << entity_desc_resp->entity_name()
+                        << " (0x" << std::setw(16) << std::hex << std::setfill('0') << end_station->entity_id()
+                        << ")" << std::endl;
+            atomic_cout << "\tEntity Index:        " << std::dec << current_entity << std::endl;
```
Comment (vronchetti) [inline controller/app/cmdline/src/cmd_line.cpp]: ```current_entity``` is a local assigned to the previous entity index - we want to print the new one.

File: controller/app/cmdline/src/cmd_line.cpp [hunk 2]
```diff
@@ -1382,30 +1382,34 @@ int cmd_line::do_select(uint32_t new_end_station, uint16_t new_entity, uint16_t
...
+            atomic_cout << "\tConfiguration Index: " << std::dec << current_config << std::endl;
```
Comment (vronchetti) [inline controller/app/cmdline/src/cmd_line.cpp]: same for ```current_config```

File: controller/app/cmdline/src/cmd_line.cpp (hunk 1 above)
Comment (ahogen) [inline controller/app/cmdline/src/cmd_line.cpp]: Whoops! Thanks!

**Comment by andrew-elder** [comment]
I'm going to give XMOS and @fchin 24 hours to comment before accepting this pull request.

**Pull request merged**